pkg/services/ngalert/notifier/channels/opsgenie.go

```diff
@@ -220,3 +249,22 @@ func (on *OpsgenieNotifier) sendDetails() bool {
 func (on *OpsgenieNotifier) sendTags() bool {
 	return on.SendTagsAs == OpsgenieSendTags || on.SendTagsAs == OpsgenieSendBoth
 }
+
+// ExtractAlertKey gets the alert key based on the label set.
+func ExtractAlertKey(alert *types.Alert) notify.Key {
+	var sb strings.Builder
+
+	labels := make([]string, len(alert.Labels))
+	i := 0
+	for key := range alert.Labels {
+		labels[i] = string(key)
```

Comment:
> This can be labels := make([]string, 0, len(alert.Labels))

and here this can be `labels = append(labels, string(key))`. The `i := 0` and `i++` is then no longer needed.